src/eh1/io.rs

```diff
@@ -0,0 +1,684 @@
+            panic!("response longer than read buffer for io::read");
+        }
+
+        let len = std::cmp::min(buffer.len(), transaction.response.len());
+        buffer[..len].copy_from_slice(&transaction.response[..len]);
+
+        match transaction.expected_err {
+            Some(err) => Err(err),
+            None => Ok(len),
+        }
+    }
+}
+
+impl Seek for Mock {
+    fn seek(&mut self, pos: SeekFrom) -> Result<u64, Self::Error> {
+        let transaction = self.next().expect("no expectation for io::seek call");
+
+        if let Mode::Seek(expected_pos) = transaction.expected_mode {
+            assert_eq!(expected_pos, pos, "io::seek unexpected mode");
+
+            let ret_offset: u64 = u64::from_be_bytes(transaction.response.try_into().unwrap());
+            match transaction.expected_err {
+                Some(err) => Err(err),
+                None => Ok(ret_offset),
+            }
+        } else {
+            panic!(
+                "expected seek transaction, but instead encountered {:?}. io::seek unexpected mode",
+                transaction.expected_mode
+            );
+        }
+    }
+}
+
+impl WriteReady for Mock {
+    fn write_ready(&mut self) -> Result<bool, Self::Error> {
+        let transaction = self
+            .next()
+            .expect("no expectation for io::write_ready call");
+
+        match transaction.expected_mode {
+            Mode::WriteReady(ready) if transaction.expected_err.is_none() => Ok(ready),
+            Mode::WriteReady(_) if transaction.expected_err.is_some() => {
+                Err(transaction.expected_err.unwrap())
+            }
+            _ => panic!(
+                "expected write_ready transaction, but instead encountered {:?}. io::write_ready unexpected mode",
+                transaction.expected_mode
+            ),
+        }
+    }
+}
+
+impl ReadReady for Mock {
+    fn read_ready(&mut self) -> Result<bool, Self::Error> {
+        let transaction = self.next().expect("no expectation for io::read_ready call");
+
+        match transaction.expected_mode {
+            Mode::ReadReady(ready) if transaction.expected_err.is_none() => Ok(ready),
+            Mode::ReadReady(_) if transaction.expected_err.is_some() => {
+                Err(transaction.expected_err.unwrap())
+            }
+            _ => panic!(
+                "expected read_ready transaction, but instead encountered {:?}. io::read_ready unexpected mode",
+                transaction.expected_mode
+            )
+        }
+    }
+}
+
+impl BufRead for Mock {
+    fn fill_buf(&mut self) -> Result<&[u8], Self::Error> {
+        let transaction = self.next().expect("no expectation for io::fill_buf call");
+        assert_eq!(
+            transaction.expected_mode,
+            Mode::FillBuff,
+            "io::fill_buf unexpected mode"
+        );
+
+        let response_vec = transaction.response;
+        self.set_mock_data(Some(response_vec));
```

Comment:
Since this variable is only used once, I think it can be inlined.

```suggestion
        self.set_mock_data(Some(transaction.response;));
```